src/plugins/extras/extras.js

```diff
@@ -48,6 +52,21 @@
         return text.split( SLIDE_SEPARATOR );
     };
 
+    const guessSlideTitle = function( text ) {
+        for ( var line of text.split( "\n" ) ) {
+            line = line.trim( );
+            if ( line.length > 0 ) {
+                if ( line.length <= MAX_TITLE_LEN ) {
+                    return line;
+                }
+
+                // The first non-blank is too long to be a title
+                return "";
```

Comment:
Why not cut the line to 40 char?

Reply:
For a long sentence, I don't know whether it's suitable to keep the first part.  But maybe it is better than `page-n`? I'll update the PR.